src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -419,7 +419,7 @@ class VM_Version_StubGenerator: public StubCodeGenerator {
     __ movl(Address(rsi, 8), rcx);
     __ movl(Address(rsi,12), rdx);
 
-#ifndef PRODUCT
+#if !defined(PRODUCT) && defined(_LP64)
```

Comment:
Why it is still under `!PRODUCT`?  You need to check if OS supports it in product VM too. Or I missing something?

Reply:
Idea is to enable the APX support for product builds only after completion of planned features listed in JDK-8329030.

I have moved to [appropriate place](https://github.com/openjdk/jdk/pull/19042/files#diff-6ed856c57ddbe33e49883adb7c52ec51ed377e5f697dfd6d8bea505a97bfc5a5R3186)

Reply:
Thank you for moving `PRODUCT` check and adding comment.

New save/restore methods should be under `#ifdef _LP64`. 32-bit build is broken again because of that.